lib/tools/system-calls.js

```diff
@@ -85,8 +84,17 @@ systemCallMethods.getBinaryFromSdkRoot = _.memoize(async function (binaryName) {
                     path.resolve(this.sdkRoot, "tools", "bin", binaryName)];
   // get subpaths for currently installed build tool directories
   let buildToolDirs = await getDirectories(path.resolve(this.sdkRoot, "build-tools"));
+  const buildDirsByMTime = {};
+  for (const buildToolDir of buildToolDirs) {
+    const fullPath = path.resolve(this.sdkRoot, "build-tools", buildToolDir);
+    const {mtime} = await fs.stat(fullPath);
+    buildDirsByMTime[mtime] = buildToolDir;
+  }
   // the newest version goes first
-  buildToolDirs.sort(semver.rcompare);
+  const sortedMtimes = _.keys(buildDirsByMTime).sort().reverse();
```

Comment:
This is sorting by `mtime`, which will be sorted _alphabetically_, not according to time. For an original object like
```js
{ 'Thu Jan 04 2018 12:38:38 GMT-0500 (EST)': '23.0.1',
  'Thu Jan 04 2018 12:04:22 GMT-0500 (EST)': '25.0.0',
  'Tue Dec 19 2017 14:33:17 GMT-0500 (EST)': '26.0.2',
  'Tue Dec 19 2017 14:17:43 GMT-0500 (EST)': '27.0.2',
  'Fri Dec 29 2017 13:16:46 GMT-0500 (EST)': '27.0.3' }
```
the "sorted" array will be
```js
[ 'Tue Dec 19 2017 14:33:17 GMT-0500 (EST)',
  'Tue Dec 19 2017 14:17:43 GMT-0500 (EST)',
  'Thu Jan 04 2018 12:38:38 GMT-0500 (EST)',
  'Thu Jan 04 2018 12:04:22 GMT-0500 (EST)',
  'Fri Dec 29 2017 13:16:46 GMT-0500 (EST)' ]
```

Reply:
The solution is to pull `mtimeMs` off the stat object, so the sorting is on a number.

Reply:
Changed the algorithm to sort by `mtimeMs`. Ehre is the page where I was getting the info: https://nodejs.org/api/fs.html#fs_class_fs_stats